python/pyspark/sql/functions/builtin.py

```diff
@@ -262,6 +290,25 @@ def lit(col: Any) -> Column:
                 errorClass="COLUMN_IN_LIST", messageParameters={"func_name": "lit"}
             )
         return array(*[lit(item) for item in col])
+    elif isinstance(col, dict):
+        # Skip checking if the keys are column as Columns are not hashable
+        # and cannot be used as dict keys in the first place.
+        if any(isinstance(value, Column) for value in col.values()):
+            raise PySparkValueError(
+                errorClass="COLUMN_IN_DICT", messageParameters={"func_name": "lit"}
+            )
+        from pyspark.sql import SparkSession
+
+        spark = SparkSession.getActiveSession()
```

Comment:
do we require an active session here? @HyukjinKwon

Reply:
Let's use the default value of `spark.sql.pyspark.inferNestedDictAsStruct.enabled` when `SparkSession.getActiveSession` returns `None`. For Connect too.